server/issue.go

```diff
@@ -199,6 +199,78 @@ func httpAPIGetCreateIssueMetadata(ji Instance, w http.ResponseWriter, r *http.R
 	return http.StatusOK, nil
 }
 
+func httpAPIGetSearchIssues(ji Instance, w http.ResponseWriter, r *http.Request) (int, error) {
+	if r.Method != http.MethodGet {
+		return http.StatusMethodNotAllowed,
+			errors.New("Request: " + r.Method + " is not allowed, must be GET")
+	}
+
+	mattermostUserId := r.Header.Get("Mattermost-User-Id")
+	if mattermostUserId == "" {
+		return http.StatusUnauthorized, errors.New("not authorized")
+	}
+
+	jiraUser, err := ji.GetPlugin().LoadJIRAUser(ji, mattermostUserId)
+	if err != nil {
+		return http.StatusInternalServerError, err
+	}
+
+	jiraClient, err := ji.GetJIRAClient(jiraUser)
+	if err != nil {
+		return http.StatusInternalServerError, err
+	}
+
+	err = r.ParseForm()
+	if err != nil {
+		return http.StatusInternalServerError, errors.WithMessage(err, "failed to parse request")
+	}
+
+	jqlString := r.Form.Get("jql")
+
+	searchRes, resp, err := jiraClient.Issue.Search(jqlString, &jira.SearchOptions{
+		MaxResults: 50,
+		Fields:     []string{"key", "summary"},
+	})
+
+	if err != nil {
```

Comment:
for 2.1, we should refactor into a function, maybe `RespondWithJiraError(status, err, resp)`